pageserver/src/walredo.rs

```diff
@@ -308,7 +311,12 @@ impl PostgresRedoManager {
                 // than we can SIGKILL & `wait` for them to exit. By doing it the way we do here,
                 // we limit this risk of run-away to at most $num_runtimes * $num_executor_threads.
                 // This probably needs revisiting at some later point.
+                let mut wait_done = proc.stderr_logger_task_done.clone();
                 drop(proc);
+                match wait_done.wait_for(|v| *v).await {
+                    Ok(_) => {}
+                    Err(_) => todo!(),
```

Comment:
can you either fill the body or make this an unwrap with an explanation why we never would close the `stderr_logger_task_done_tx` before sending something to it? I suppose as we send something via the scope guard macro, we should be safe?

Reply:
eaf1077731632ea9029e286046a731c19346e146